### PR TITLE
[Admin] Implement string blocks management

### PR DESCRIPTION
**Pull request by pjedrzejewski**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Related tickets | - |
| License | MIT |
- Scenarios for admin
- Admin CRUD
- Fixtures
- Added PHPCR Initializer

### CODE REVIEW

File: features/cmf/managing_string_blocks/adding_string_block.feature [hunk 1]
```diff
@@ -0,0 +1,17 @@
+@managing_string_blocks
+Feature: Adding a new string block
+    In order to manage content
```
Comment (pamil) [inline features/cmf/managing_string_blocks/adding_string_block.feature]: content?

File: features/cmf/managing_string_blocks/deleting_string_block.feature [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+Feature: Deleting a string block
+    In order to remove test, obsolete or incorrect string blocks
+    As an Administrator
+    I want to be able to delete a string block
+
+    Background:
+        Given I am logged in as an administrator
+
+    @ui
+    Scenario: Deleted taxon should disappear from the registry
```
Comment (pamil) [inline features/cmf/managing_string_blocks/deleting_string_block.feature]: Not a taxon for sure 🎉

File: features/cmf/managing_string_blocks/string_block_validation.feature [hunk 3]
```diff
@@ -0,0 +1,23 @@
+@managing_string_blocks
+Feature: Static contents validation
```
Comment (pamil) [inline features/cmf/managing_string_blocks/string_block_validation.feature]: Static content :)

File: src/Sylius/Behat/Context/Setup/StringBlockContext.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+namespace Sylius\Behat\Context\Setup;
+
+use Behat\Behat\Context\Context;
+use Doctrine\Common\Persistence\ObjectManager;
+use Sylius\Behat\Service\SharedStorageInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
```
Comment (pamil) [inline src/Sylius/Behat/Context/Setup/StringBlockContext.php]: 🌮

File: src/Sylius/Behat/Context/Ui/Admin/ManagingStringBlocksContext.php [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+     * @When I add it
+     * @When I try to add it
+     */
+    public function iAddIt()
+    {
+        $this->createPage->create();
+    }
+
+    /**
+     * @Then /^I should be notified that (body|name|name) is required$/
```
Comment (pamil) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingStringBlocksContext.php]: Duplicated `name`.

File: src/Sylius/Behat/Context/Ui/Admin/ManagingStringBlocksContext.php [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+     */
+    public function theStringBlockShouldNotBeAdded($name)
+    {
+        if (!$this->indexPage->isOpen()) {
+            $this->indexPage->open();
+        }
+
+        Assert::false(
+            $this->indexPage->isSingleResourceOnPage(['name' => $name]),
+            sprintf('Static content with name %s was created, but it should not.', $name)
```
Comment (pamil) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingStringBlocksContext.php]: Static content once again 🐫

File: src/Sylius/Bundle/ContentBundle/Factory/BlockFactory.php [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        $this->blockParentPath = $blockParentPath;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createNew()
+    {
+        $block = $this->decoratedFactory->createNew();
+        $block->setParentDocument($parent = $this->documentManager->find(null, $this->blockParentPath));
```
Comment (pamil) [inline src/Sylius/Bundle/ContentBundle/Factory/BlockFactory.php]: `$parent =` is not needed.

File: src/Sylius/Behat/Page/Admin/StringBlock/CreatePageInterface.php [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Page\Admin\StringBlock;
+
+use Sylius\Behat\Page\Admin\Crud\CreatePageInterface as BaseCreatePageInterface;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Page/Admin/StringBlock/CreatePageInterface.php]: And again 🎉

File: src/Sylius/Bundle/CoreBundle/Application/Kernel.php [hunk 9]
```diff
@@ -65,6 +65,9 @@ public function registerBundles()
             new \Symfony\Cmf\Bundle\ContentBundle\CmfContentBundle(),
             new \Symfony\Cmf\Bundle\RoutingBundle\CmfRoutingBundle(),
             new \Symfony\Cmf\Bundle\MenuBundle\CmfMenuBundle(),
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/Application/Kernel.php]: Shouldn't this be deleted already? Do we use it somewhere?

Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Application/Kernel.php]: `CmfContentBundle` requires it.

**Pull request closed without merging**